pkg/interceptors/github/github.go

```diff
@@ -96,7 +129,306 @@ func (w *Interceptor) Process(ctx context.Context, r *triggersv1.InterceptorRequ
 		}
 	}
 
+	// For event types pull_request, issue_comment check github owners approval is required
+	// User can specify both event type or any one of them
+	if p.GithubOwners.Enabled {
+		ownerCheckAllowed := false
+		for _, allowedEvent := range ownersEventTypes {
+			if actualEvent == allowedEvent {
+				ownerCheckAllowed = true
+				break
+			}
+		}
+		if ownerCheckAllowed {
+			ghToken, err := w.getPersonalAccessTokenSecret(ctx, r, p)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error getting github token: %v", err)
+			}
+			if ghToken == "" && (p.GithubOwners.EnableOrgMemberCheck || p.GithubOwners.EnableRepoMemberCheck) {
+				return interceptors.Fail(codes.FailedPrecondition, "enableOrgMemberCheck or enableRepoMemberCheck is enabled but no personalAccessToken was supplied")
+			}
+
+			enterpriseBaseURL := headers.Get("X-Github-Enterprise-Host")
+			client, err := makeClient(ctx, enterpriseBaseURL, ghToken)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error making client: %v", err)
+			}
+			payload, err := parseBodyForOwners(r.Body, actualEvent)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error parsing body: %v", err)
+			}
+			allowed, err := checkOwnershipAndMembership(ctx, payload, p, client)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error checking owner verification: %v", err)
+			}
+
+			if allowed && actualEvent == "pull_request" {
+				return &triggersv1.InterceptorResponse{
+					Continue: true,
+				}
+			}
+
+			commentAllowed, err := okToTestFromAnOwner(ctx, payload, p, client)
+			if err != nil {
+				return interceptors.Failf(codes.FailedPrecondition, "error checking comments for verification: %v", err)
+			}
+			if !commentAllowed {
+				return interceptors.Fail(codes.FailedPrecondition, "owners check requirements not met")
+			}
+		}
+	}
+
 	return &triggersv1.InterceptorResponse{
 		Continue: true,
 	}
 }
+
+func (w *Interceptor) getPersonalAccessTokenSecret(ctx context.Context, r *triggersv1.InterceptorRequest, p triggersv1.GitHubInterceptor) (string, error) {
+	if p.GithubOwners.PersonalAccessToken == nil {
+		return "", nil
+	}
+	if p.GithubOwners.PersonalAccessToken.SecretKey == "" {
+		return "", fmt.Errorf("github interceptor personalAccessToken.secretKey is empty")
+	}
+	if r.Context == nil {
+		return "", fmt.Errorf("no request context passed")
+	}
+	ns, _ := triggersv1.ParseTriggerID(r.Context.TriggerID)
+	secretToken, err := w.SecretGetter.Get(ctx, ns, p.GithubOwners.PersonalAccessToken)
+	if err != nil {
+		return "", err
+	}
+	return string(secretToken), nil
+}
+
+func okToTestFromAnOwner(ctx context.Context, payload OwnersPayloadDetails, p triggersv1.GitHubInterceptor, client *gh.Client) (bool, error) {
+	if MatchRegexp(OKToTestCommentRegexp, payload.IssueCommentBody) {
```

Comment:
From the code here it seems we would allow either owners from an owners file or if an owner adds a ok-to-test comment - we should document this ok-to-test behavior in the docs